README.adoc

```diff
@@ -0,0 +1,402 @@
+
+**Writings around FP**
+
+* https://github.com/stoeffel/awesome-fp-js[Awesome FP JS] list
+* http://blog.wolksoftware.com/the-rise-of-functional-programming-and-the-death-of-angularjs[The rise of functional programming & the decline of Angular 2.0]
+
+#### `import` (https://github.com/benmosher/eslint-plugin-import[↗])
+Provides rules that help prevent import bugs and enforces style.
+
+#### `vue` (https://github.com/vuejs/eslint-plugin-vue[↗])
+Currently, the sole purpose of having this plugin is to parse https://vuejs.org/v2/guide/single-file-components.html[`.vue` files]. Also, it https://github.com/vuejs/eslint-plugin-vue/issues/1[does not support] eslint's `--fix` feature.
+
+## Extends
+
+#### Rules of AirBnB (https://github.com/airbnb/javascript[↗])
+
+We extend the AirBnB rules for historical reasons. Our configuration will continue to evolve and may not be based on it one day if we eventually disable or adjust too much of it via overrides.
+
+
+
+
+## Rules
+
+### `semi`
+
+Forbidden. +
+
+Refactoring is harder. +
+
+The act of changing/moving code  is made tedious by constantly baby sitting removal/addition of semicolons. For example: transitioning between `() => (...)` / `() => {...}`, moving the last `.then`/`.catch`/ `.foo` in a pipeline up or appending to it, etc.. +
+
+Reading and writing is harder. +
+
+JavaScript is a syntactically verbose language. Semicolons make it worse yet. Reading code with semicolons is objectively more fatiguing than code without given the reduction in visual noise. For writing, a day of coding with semicolons wears more on the fingers/hand than code without. +
+
+Further reading +
+
+* http://stackoverflow.com/questions/2846283/what-are-the-rules-for-javascripts-automatic-semicolon-insertion-asi[ASI rules]
+* http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding[An open letter to JavaScript leaders regarding Semicolons]
+* http://inimino.org/~inimino/blog/javascript_semicolons[JavaScript Semicolon Insertion; Everything you need to know]
+* http://mislav.net/2010/05/semicolons/[Semicolons in JavaScript are optional]
+
+**Fail**
+```js
+it("foobar", () => {
+  assert(1, foo(1));
+});
+```
+**Pass**
+```js
+it("foobar", () => {
+  assert(1, foo(1))
+})
+```
+
+### `quotes`
+
+Double. +
+
+Some languages treat single/double as different types (Java, Haskell, PureScript, ...), don't even have single quotes (Clojure), or idiomatically use double (HTML). It is therefore better (Assuming a polyglot programmer) for habit building and retention to use double quotes as well in JavaScript.
+
+**Fail**
+```js
+import Foo from 'Foo'
+
+console.log('Foo is: %j', Foo)
+```
+
+**Pass**
+```js
+import Foo from "Foo"
+
+console.log("Foo is: %j", Foo)
+```
+
+### `no-multiple-empty-lines`
+
+Up to three allowed. Two empty lines are not enough to clearly partition major sections of a module (e.g. after all `import ...`).
```

Comment:
I personally would like to keep it at one line, but if there's a consensus then we can use three line, but this would allow us to have multiple lines everywhere.